### PR TITLE
Pin requests to 2.10.0 for mrest compatibility

### PR DESCRIPTION
**Pull request by mellenburg**
Azure broke recently with an exception in internal API calls

### CODE REVIEW

**Comment by cmaloney** [comment]
https://github.com/dcos/dcos/blob/master/packages/dcos-image-deps/buildinfo.json#L34 -- want to make sure to match that version with any pin.

**Comment by mellenburg** [comment]
@cmaloney I found the real source of the issue requests 2.11 came out and it will now raise an exception if any header fields arent passed as string (2.10 simply accepts it and will cast as string without complaining) and mrest passes content-length as an int. mrest added a fix yesterday that will be in 0.4.1. Just a note for when we try to upgrade to 2.11 that stuff may break

**Comment by cmaloney** [comment]
Can we make this 2.10.0 so it matches: https://github.com/dcos/dcos/blob/master/packages/python-requests/buildinfo.json#L5?

With that can ship

File: packages/python-requests/build [hunk 1]
```diff
@@ -3,4 +3,5 @@ source /opt/mesosphere/environment.export
 export LIB_INSTALL_DIR="$PKG_PATH/lib/python3.4/site-packages"
 mkdir -p "$LIB_INSTALL_DIR"
 
-pip3 install --no-deps --install-option="--prefix=$PKG_PATH" --root=/ /pkg/src/$PKG_NAME
+cd /pkg/src/$PKG_NAME
+PYTHONPATH=$PYTHONPATH:$LIB_INSTALL_DIR easy_install --prefix=$PKG_PATH .
```
Comment (cmaloney) [inline packages/python-requests/build]: Why do we need to do this? Why isn't the pypi one good enough?

This also isn't one of the standard python install scripts we have with this change. `easy_install` is also deprecated I believe. Things are supposed to always use pip + wheel.

File: setup.py [hunk 2]
```diff
@@ -48,9 +48,10 @@ def get_advanced_templates():
         'docopt',
         'passlib',
         'pyyaml',
-        'requests',
+        'requests==2.10',
```
Comment (cmaloney) [inline setup.py]: please just pin this to 2.10.0 directly and not random requests commit zip.

File: test_util/azure_test_driver.py [hunk 3]
```diff
@@ -55,7 +55,7 @@ def run():
     # tenant_id = os.environ.get('AZURE_TENANT_ID')
     # client_id = os.environ.get('AZURE_CLIENT_ID')
     # client_secret = os.environ.get('AZURE_CLIENT_SECRET')
-    group_name = 'tesing' + ''.join(random.choice('01234567890abcdef') for n in range(10))
+    group_name = 'testing' + ''.join(random.choice('01234567890abcdef') for n in range(10))
```
Comment (cmaloney) [inline test_util/azure_test_driver.py]: this is an independent change, another pr is pretty simple / easy.

**Comment by mellenburg** [comment]
@cmaloney I mistakenly took "so that it matches" to be 'the guaranteed same bits" as opposed to 'add .0'

**Comment by cmaloney** [comment]
@mellenburg Sorry about that. In theory the "add the .0" will cause pip to install exactly that version using the same url, with a lot less work.

**Comment by cmaloney** [comment]
🚢

**Pull request merged**